drivers/net/wireless/mediatek/mt76/mt76_connac_mac.c

```diff
@@ -13,6 +13,9 @@ int mt76_connac_pm_wake(struct mt76_phy *phy, struct mt76_connac_pm *pm)
 	if (mt76_is_usb(dev))
 		return 0;
 
+	if (test_bit(MT76_STATE_SUSPEND, &phy->state))
+		return 0;
+
 	cancel_delayed_work_sync(&pm->ps_work);
```

Comment:
@objelf is it enough to just return after cancel_delayed_work_sync() if the device is not running (MT76_STATE_RUNNING not set)? Doing so we can drop the MT76_STATE_WOW patch, what do you think?